### ISSUE TITLE
RFE : What happens if you manually add a waypoint to a route?

### ISSUE THREAD

**Issue by Gilletarom** [rgleason/weather_routing_pi#129]
Hello Dave,
During my last navigation, the route proposed by WR passed over a shoal, ie an area where there was not enough water for my boat. This was due to an error of observation on my part.

During the navigation, I therefore changed the route locally. I manually added a waypoint to divert the route and pass this shoal.
As a result, all the distance calculations and the calculations on the expected times have become erroneous.

Question: In the current state of WR (1.15.2) how does this affect distance calculations and crossing times? Can we trust what appears on the route in the OpenCPN Route Manager? AND on the track recorded in the track manager?

**Comment by rgleason** [comment]
Please see RFE #11  https://github.com/rgleason/weather_routing_pi/issues/84
to add a new constraint for a WPT to be passed to port or starboard.

**Comment by Gilletarom** [comment]
> Is this after or before you've exported it? I expect it is after export to Routes.
> 

Oui, j'était arrivé en fin de journée près du port final et la marée, à marée basse. La route crée le matin, et exportée, directement, en tant que route passait entre une tourelle et un petit phare. dans ce bras d'eau, il y avait des  eaux fonds marqués à 1,20 mètre et 2,36 mètres. Faute d'avoir la certitude que la route ne passait pas sur un de ces deux zones, j'ai préféré ajouter un waypoint à la route, et faire passer la route au nord de la tourelle.

Le nouveau waypoint est apparu en bleu, avec la même icône que traditionnellement. J'ai même pu modifier la route plusieurs fois. Aucun crash. 

Hélas, suite à une fausse manoeuvre, j'ai effacé la route  ... Grrrr ....  Je tente d'envoyer la trace enregistrer dans OpenCPN, mais Github n'accepte pas les fichier gpx donc ????
![Trace_2023-08-21_Lundi_St-Cast_St-Quay gpx](https://github.com/rgleason/weather_routing_pi/assets/6445543/195b8274-a2d3-4f9c-b20e-8601803abf64)

Nota bene : Vous pouvez comprendre à la lecture de ce texte pourquoi j'ai pu proposer XXXXXX

Gilletarom

**Comment by Gilletarom** [comment]
I had forgotten that I had sent the route, after modification, to the pair of OpenCPN on my tablet! And don, I found the road by reference on my PC.
![Route_2023-08-21_Lundi_St-Cast_St-Quay_Meteoroute_modifiee gpx](https://github.com/rgleason/weather_routing_pi/assets/6445543/e4b98485-6cc3-499e-860e-5d2b993df548)

But, it is a gpx file and Github does not accept this type of file.

I can provide it by email.


Gilletarom

**Comment by Gilletarom** [comment]
her e joined a screencopy of the route after that I add a waypoint :

![Screenshot_20230825-224429](https://github.com/rgleason/weather_routing_pi/assets/6445543/90ab3690-15f3-48f8-a2bf-9daca83f4250)

**Comment by rgleason** [comment]
@Gilletarom This issue appears to be about an already exported gpx file and how it cannot be successfully imported into grib_pi????

This does not make sense to me. Gpx files are imported by Opencpn Routes.

This is an OpenCpn Issue.
Please close this and open a new issue under Opencpn, only if it is valid and still a bug. I note this is 2 years old now.

Now closing this. If you disagree, please go ahead and comment why  and reopen.

Thanks Rick

**Comment by rgleason** [comment]
@sebastien-rosset  What do you think?
I believe Gille had exported the route to OpenCPN.

@Gilletarom Can you still repeat this issue and provide us with step by step instructions? This is 2yrs ild.

**Comment by rgleason** [comment]
@Gilletarom  I have tried to transfer this Issue to OpenCPN for you and cannot do it. So the only alternative is for you to make a new issue there.  As far as I can determine, this Issue is not related to Weather_routing as the routing has been exported to  a route, although you do not make that clear in your description.
